### PR TITLE
[SPARK-42732][PYTHON][CONNECT] Update spark connect session `getOrCreate` behavior to check existing global `_active_spark_session` first

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

Update spark connect session `getOrCreate` behavior to check existing global `_active_spark_session` first, if existing, return it.

Spark connect ML needs this API to get active session in some cases (e.g. fetching model attributes from server side).

### Why are the changes needed?

Manually.

### Does this PR introduce _any_ user-facing change?

Yes. Implemented `pyspark.sql.connect.session.SparkSession.getActiveSession` API. 

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @HyukjinKwon @zhengruifeng

File: python/pyspark/sql/connect/session.py [hunk 1]
```diff
@@ -119,7 +122,11 @@ def enableHiveSupport(self) -> "SparkSession.Builder":
             raise NotImplementedError("enableHiveSupport not implemented for Spark Connect")
 
         def getOrCreate(self) -> "SparkSession":
-            return SparkSession(connectionString=self._options["spark.remote"])
```
[the remark this answers is not in the document]
Comment (WeichenXu123) [reply python/pyspark/sql/connect/session.py:122]: I don't think so.
pyspark.sql.SparkSession also does not add lock when calling getOrCreate

**Comment by hvanhovell** [comment]
@WeichenXu123 in what case won't Spark Connect ML have access to the session?

**Comment by WeichenXu123** [comment]
> @WeichenXu123 in what case won't Spark Connect ML have access to the session?

For some APIs, like `estimator.fit(dataset)`, `model.transform(dataset)`, we can get session from the input spark dataframe, in other cases e.g. get a model attribute, we have no input dataframe, so we need `getActiveSession` to get the session and then send getting attribute requests to server side.

**Comment by WeichenXu123** [comment]
CC @HyukjinKwon @grundprinzip Thoughts ? ML use cases requires the global session and the session must be the same during the whole ML program execution.

**Review by zhengruifeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
LGTM but would be great to add a couple of comment there ..... it's hard to follow why especially if you don't have some background about ML implementation ..

File: python/pyspark/sql/connect/session.py [hunk 2]
```diff
@@ -427,7 +436,7 @@ def stop(self) -> None:
 
     @classmethod
     def getActiveSession(cls) -> Any:
-        raise NotImplementedError("getActiveSession() is not implemented.")
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/session.py:430]: Can we don't expose this but use `_active_spark_session` in ML instead? The behaviour of this implementation is different from regular Spark (e.g., 1. actions like `collect` should set the active Spark Session but this PR doesn't. 2. options provided should be set even when there's an active session)

**Review by HyukjinKwon — COMMENTED** [review]
I am good with this workaround to unblock ML implementation for now ... as it at least doesn't add a new API yet.

**Comment by WeichenXu123** [comment]
merged to master

**Comment by HyukjinKwon** [comment]
@WeichenXu123 mind fixing the PR title? This PR doesn't add the getActiveSession method

**Comment by WeichenXu123** [comment]
> @WeichenXu123 mind fixing the PR title? This PR doesn't add the getActiveSession method

Updated.
